### PR TITLE
Improve result sorting

### PR DESCRIPTION
**Pull request by MrPetovan**
I finally got around to implement sorting by last activity. In the process, I also sort on number of filled profile fields so that the most verbose results show up first.

The number of filled profile fields is weighted:
- Description earns 4 points
- Any number of Tags earns 2 points
- Any number of location fields earns 1 point

The profile card now show the region if it isn't the same string as the locality.

Thanks to Adam Clarke for the nudge.

### CODE REVIEW

File: mod/search.php [hunk 1]
```diff
@@ -51,38 +51,40 @@ function search_content(App $a)
...
 		$total = $r[0]['total'];
 		$a->set_pager_total($total);
 	}
 
-	if ($alpha) {
-		$order = ' ORDER BY `name` ASC ';
-	} else {
-		$order = ' ORDER BY `updated` DESC, `id` DESC ';
-	}
+	$query = "SELECT *
```
Comment (tobiasd) [inline mod/search.php]: The indentation looks wrong in the github diff. Could you have a look?

Comment (MrPetovan) [reply mod/search.php]: It's tabs everywhere, what's the matter?

Comment (MrPetovan) [reply mod/search.php]: It's tabs everywhere, what are you seeing wrong?

File: mod/directory.php [hunk 2]
```diff
@@ -62,7 +62,7 @@ function directory_content(App $a)
 	if ($alpha) {
 		$order = ' ORDER BY `name` ASC ';
 	} else {
-		$order = ' ORDER BY `updated` DESC, `id` DESC ';
+		$order = ' ORDER BY `filled_fields` DESC, `last_activity` DESC, `updated` DESC';
```
Comment (annando) [inline mod/directory.php]: Wouldn't it be better if the sorting would be by the month of the last update and then the number of filled fields? Sorting after the filled fields at first could put someone at first who had filled out all fields - but hadn't been active for years.

Comment (MrPetovan) [reply mod/directory.php]: The last_actitivy is `YYYY-WW` with WW being the week number in the year, so we can't sort on the month if it made any sense. Besides, this sorting is only to improve the visual impression of the default display of the directory. The first page will always have filled profiles with recent activity. What happens in the subsequent pages is way less relevant.

File: mod/search.php (hunk 1 above)
Comment (MrPetovan) [reply mod/search.php]: If you're talking about the SQL query strings, it's intentional, a old habit I took when I was outputting the queries for debug and I loathed the extra white spaces in the debug display. So I started pulling the queries left to remove the tabs/spaces.

File: mod/directory.php (hunk 2 above)
Comment (annando) [reply mod/directory.php]: Ah, yeah. We already compacted the data. So possibly we should sort by "last_activity" and "filled_fields" instead.

Comment (MrPetovan) [reply mod/directory.php]: And then you'd have all the spam bots without any profile info first. Believe me, I thought about it, I tried different orders, and this was the most visually gratifying.

Comment (annando) [reply mod/directory.php]: Okay. Since we can change that very fast (and since there aren't that many installations), we can test it out.

**Comment by annando** [comment]
P.S.: Could you have a look at https://github.com/friendica/dir/issues/15 as well?

**Comment by AlfredSK** [comment]
@MrPetovan Is it sufficient to do a 'git pull' to update the directory?

**Comment by MrPetovan** [comment]
Nope, there are changes to the DB structure that aren't applied automatically. Please look at the changes to `dfrndir.sql` in this PR.

**Comment by MrPetovan** [comment]
Nope, there never was for directory.

**Pull request merged**